shared/events.js

```diff
@@ -38,11 +38,15 @@ export class Hashable {
   }
   // https://flowtype.org/docs/classes.html#this-type
   static fromObject (obj, hash): this {
-    var instance = new this()
-    instance._obj = obj
+    var instance = this.fromJSON(obj)
     if (instance.toHash() !== hash) throw Error(`hash obj: ${instance.toHash()} != hash: ${hash}`)
     return instance
   }
+  static fromJSON (json): this {
+    var instance = new this()
+    instance._obj = json
+    return instance
+  }
```

Comment:
`fromJSON` explicitly musn't exist in order to mandate data-integrity and quickly catch bugs. It was a conscious decision to require the `hash` to be passed along with the object in `fromObject`, which is the safe `fromJSON` that you seek.

Reply:
This is needed in order to restore an entry dataType from its serialized form back into a full blown object to be used in a transaction they way it was set. This does not circumvent the integrity check in any of the other part of the applicaiton and the integrity check is not needed in the transactions deserialization.